openid/src/main/java/fish/payara/security/openid/AccessTokenIdentityStore.java

```diff
@@ -0,0 +1,113 @@
+ *  file that accompanied this code.
+ *
+ *  Modifications:
+ *  If applicable, add the following below the License Header, with the fields
+ *  enclosed by brackets [] replaced by your own identifying information:
+ *  "Portions Copyright [year] [name of copyright owner]"
+ *
+ *  Contributor(s):
+ *  If you wish your version of this file to be governed by only the CDDL or
+ *  only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *  elects to include this software in this distribution under the [CDDL or GPL
+ *  Version 2] license."  If you don't indicate a single choice of license, a
+ *  recipient has the option to distribute your version of this file under
+ *  either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *  its licensees as provided above.  However, if you add GPL Version 2 code
+ *  and therefore, elected the GPL Version 2 license, then the option applies
+ *  only if the new code is made subject to such option by the copyright
+ *  holder.
+ */
+
+package fish.payara.security.openid;
+
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.security.enterprise.identitystore.CredentialValidationResult;
+import javax.security.enterprise.identitystore.IdentityStore;
+
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.BadJWTException;
+import fish.payara.security.openid.api.AccessTokenCallerPrincipal;
+import fish.payara.security.openid.api.OpenIdConstant;
+import fish.payara.security.openid.controller.TokenClaimsSetVerifier;
+import fish.payara.security.openid.domain.AccessTokenImpl;
+import fish.payara.security.openid.domain.OpenIdConfiguration;
+import fish.payara.security.openid.domain.OpenIdContextImpl;
+
+@ApplicationScoped
+public class AccessTokenIdentityStore implements IdentityStore {
+    private static final Logger LOGGER = Logger.getLogger(AccessTokenIdentityStore.class.getName());
+
+    @Override
+    public Set<ValidationType> validationTypes() {
+        return Collections.singleton(ValidationType.VALIDATE);
+    }
+
+    @Inject
+    OpenIdContextImpl context;
+
+    @SuppressWarnings("unused")
+    public CredentialValidationResult validate(AccessTokenCredential credential) {
+        try {
+            AccessTokenImpl accessToken = AccessTokenImpl.forBearerToken(credential.getConfiguration(),
+                    credential.getAccessToken(),
+                    new BearerVerifier(credential.getConfiguration()));
+            context.setAccessToken(accessToken);
+            // for setClaims we'd need to invoke userinfo. That should be lazy unless required
+            // credential.getConfiguration().getClaimsConfiguration().getCallerNameClaim() might be better to use,
```

Comment:
Check can be added for custom caller name claim:
````
            if (accessToken.getClaim(credential.getConfiguration().getClaimsConfiguration().getCallerNameClaim()) != null) {
                context.setCallerName((String) credential.getConfiguration().getClaimsConfiguration().getCallerNameClaim());
            } else {
                context.setCallerName((String) accessToken.getClaim(OpenIdConstant.SUBJECT_IDENTIFIER));
            }
````

Reply:
Incorporated in e7dc72261c4e632f2ebfcf3cf04130aeb7fe04f3